uilayer.cpp

```diff
@@ -515,15 +515,22 @@ void ui_layer::render_background()
     if (m_background == 0)
         return;
 
-    ImVec2 size = ImGui::GetIO().DisplaySize;
+    ImVec2 display_size = ImGui::GetIO().DisplaySize;
+    ImVec2 image_size;
+    ImVec2 start_position;
+    ImVec2 end_position;
+
     opengl_texture &tex = GfxRenderer->Texture(m_background);
     tex.create();
 
-    ImGui::SetNextWindowPos(ImVec2(0, 0));
-    ImGui::SetNextWindowSize(size);
-    ImGui::Begin("Logo", nullptr,
-                 ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoScrollbar | ImGuiWindowFlags_NoScrollWithMouse |
-                     ImGuiWindowFlags_NoBringToFrontOnFocus);
-    ImGui::Image(reinterpret_cast<void *>(tex.id), size, ImVec2(0, 1), ImVec2(1, 0));
-    ImGui::End();
+    // Get the scaling factor based on the image aspect ratio vs. the display aspect ratio.
+    float scale_factor = (tex.width() / tex.height()) > (display_size.x / display_size.y)
+        ? display_size.y / tex.height()
+        : display_size.x / tex.width();
+    
+    image_size = ImVec2((int)(tex.width() * scale_factor), (int)(tex.height() * scale_factor));
+    start_position = ImVec2((display_size.x - image_size.x) / 2, (display_size.y - image_size.y) / 2);
+    end_position = ImVec2(image_size.x + start_position.x, image_size.y + start_position.y);
+
+    ImGui::GetBackgroundDrawList()->AddImage((ImTextureID)tex.id, start_position, end_position, ImVec2(0, 1), ImVec2(1, 0));
```

Comment:
The old code used `ImGui::Image()`, but this creates an ImGui window. Even with all of the window dressings hidden, a border will still show around the image, because it's still in a window. Using `GetBackgroundDrawList()->AddImage` will draw just the image without the border.